lib/perl/Genome/Model/Tools/Transcriptome/ErccMapUnaligned.pm

```diff
@@ -0,0 +1,358 @@
+            is => 'FilePath',
+            doc => 'An aligned BAM file to a reference genome without the ERCC transcripts spiked in.',
+        },
+        ercc_fasta_file => {
+            is => 'FilePath',
+            doc => 'The FASTA format sequence for all ERCC transcripts.',
+            example_values => ['/gscmnt/gc13001/info/model_data/jwalker_scratch/ERCC/ERCC92.fa'],
+        },
+        ercc_spike_in_file => {
+            is => 'Text',
+            doc => 'The control analysis file provided by Agilent for the ERCC spike-in.',
+            example_values => ['/gscmnt/gc13001/info/model_data/jwalker_scratch/ERCC/ERCC_Controls_Analysis.txt'],
+        },
+        ercc_spike_in_mix => {
+            is => 'Integer',
+            doc => 'The expected ERCC spike-in mix.',
+            valid_values => [ '1', '2'],
+        },
+        samtools_version => {
+            is => 'Text',
+            doc => 'The version of samtools to run analysis.',
+            example_values => ['1.2'],
+            default_value => '1.2',
+            is_optional => '1',
+        },
+        samtools_max_mem => {
+            is => 'Integer',
+            doc => 'The max memory required by samtools (in bytes).',
+            example_values => ['14000000000'],
+            default_value => '14000000000',
+            is_optional => '1',
+        },
+        bowtie2_version => {
+            is => 'Text',
+            doc => 'The version of bwa to run analysis.',
+            example_values => ['2.1.0'],
+            default_value => '2.1.0',
+            is_optional => '1',
+        },
+    ],
+    has_output => [
+        pdf_file => {
+            is => 'Text',
+            doc => 'The output PDF with histograms and linearity plot.',
+            default_value => 'output.pdf',
+            is_optional => '1',
+        }
+    ],
+};
+
+sub help_detail {
+    return <<EOS
+Compare the abundance of an ERCC spike-in with a known concentration.
+EOS
+}
+
+sub execute {
+    my $self = shift;
+    
+    my $ercc_bowtie_index = $self->create_ercc_bowtie_index();
+
+    my $remapped_bam = $self->generate_remapped_bam(
+        index => $ercc_bowtie_index,
+        input_bam => $self->bam_file,
+    );
+
+    $self->index_bam($remapped_bam);
+    my $idxstats = $self->generate_idxstats($remapped_bam);
+    my $tsv = $self->generate_tsvfile($idxstats);
+    $self->save_tsv_stats($tsv);
+
+    $self->run_analysis_script($tsv);
+
+    return 1;
+}
+
+sub _bin_dir {
+    return Path::Class::Dir->new("/usr/bin");
+}
+
+sub samtools {
+    my $self = shift;
+    my $version = $self->samtools_version;
+#    my $samtools_path =
+#      Genome::Model::Tools::Sam->path_for_samtools_version($version);
+
+    my $samtools_path = $self->_bin_dir->file('samtools1.2');
+
+    return $samtools_path;
+}
+
+sub bowtie2_align {
+    my $self = shift;
+    my $version = $self->bowtie2_version;
+    my $bowtie2_align = $self->_bin_dir->file('bowtie2-align' . $version);
```

Comment:
I did checkout `Genome::Model::Tools::Bowtie::path_for_bowtie_version`, but it seems that the version/installation scheme has changed for newer versions of bowtie2 and the method doesn't have any real consistent way of handling it yet.  I see different types of bowtie2 installations in-house:`bowtie2-align`, `bowtie2-align2.0.0-beta7`, `bowtie2-align2.1.0`, and `bowtie22.1.0`.  I opted, based on looking at this, to place my own "homebrew"-ed thing for my immediate QC task at hand. It wasn't clear to me what the bigger picture regarding bowtie2 installations and how to best update that method.